src/components/common/text/CryptoText.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import { useCryptoText } from '../../../hooks/useCryptoText'
+import { getDisplayDenominationFromState, getExchangeDenominationFromState } from '../../../selectors/DenominationSelectors'
+import { useDispatch, useSelector } from '../../../types/reactRedux'
+import { fixFiatCurrencyCode, getDenomFromIsoCode, zeroString } from '../../../util/utils'
+
+type Props = {
+  walletId: string,
+  tokenId?: string,
+  nativeAmount?: string
+}
+
+/**
+ * Returns a cleaned crypto amount string. If no nativeAmount is given, use the
+ * wallet balance.
+ **/
+export const CryptoText = ({ walletId, tokenId, nativeAmount }: Props) => {
+  const dispatch = useDispatch()
+  const currencyInfo = useSelector(state => state.core.account.currencyWallets[walletId].currencyInfo)
+  const fiatCurrencyCode = useSelector(state => state.core.account.currencyWallets[walletId].fiatCurrencyCode)
+  const tokenOrNativeCode = tokenId ?? currencyInfo.currencyCode
+  const exchangeRates = useSelector(state => state.exchangeRates)
+  const nativeAmountOrBalance = useSelector(state => nativeAmount ?? state.core.account.currencyWallets[walletId].balances[tokenOrNativeCode] ?? '0')
```

Comment:
Do not hook up to the balance in this way. There are two problems here:

1. The balances array is indexed by currencyCode, not tokenId. I have an outstanding task to create a `tokenBalances` property on the wallet which is indexed by tokenId. In the meantime, we have to convert the tokenId to a currencyCode before using it.
2. The balances can change dynamically. You have to combine a `useState` hook with a subscription to track it properly.

Therefore, we should just make `nativeAmount` mandatory, and not bother with any of this complexity at this level.